File: examples/typescript/basic-agent.ts
/**
 * Basic Agent Example - Open Responses API
 *
 * Demonstrates the simplest form of an agent using Open Responses
 * via the HuggingFace Inference Providers router.
 *
 * Usage:
 *     npm install openai
 *     export HF_TOKEN=your-token
 *     npx ts-node basic-agent.ts
 */

import OpenAI from "openai";

// Configure client with HuggingFace router endpoint
const client = new OpenAI({
  baseURL: "https://router.huggingface.co/v1",
  apiKey: process.env.HF_TOKEN,
});

/**
 * Create a basic agent request to Open Responses API
 */
async function createBasicAgent(
  model: string,
  input: string,
  instructions?: string
) {
  console.log("=".repeat(60));
  console.log("BASIC AGENT EXAMPLE");
  console.log("=".repeat(60));
  console.log(`Model: ${model}`);
  console.log(`Input: ${input}`);
  console.log("=".repeat(60));

  const response = await client.responses.create({
    model,
    instructions: instructions || "You are a helpful assistant.",
    input,
  });

  console.log(`\nResponse ID: ${response.id}`);
  console.log(`Model: ${response.model}`);

  // Use the convenience helper for simple text output
  console.log("\n--- Output Text (convenience helper) ---");
  console.log(response.output_text);

  // Or iterate through all output items for more detail
  console.log("\n--- All Output Items ---");
  for (const item of response.output) {
    switch (item.type) {
      case "reasoning":
        // @ts-ignore - content may exist on reasoning items
        console.log(`[REASONING] ${item.content || "[no content]"}`);
        break;
      case "message":
        // @ts-ignore - content exists on message items
        console.log(`[MESSAGE] ${item.content}`);
        break;
      default:
        console.log(`[${item.type.toUpperCase()}]`, item);
    }
  }

  // Token usage
  console.log("\n--- Usage ---");
  console.log(`Input tokens: ${response.usage?.input_tokens}`);
  console.log(`Output tokens: ${response.usage?.output_tokens}`);

  return response;
}

// Main execution
async function main(): Promise<void> {
  // Model with provider suffix - using Groq for fast inference
  const model = process.env.MODEL || "moonshotai/Kimi-K2-Instruct-0905:groq";

  if (!process.env.HF_TOKEN) {
    console.error("Error: HF_TOKEN environment variable required");
    process.exit(1);
  }

  console.log(`Using model: ${model}`);
  console.log(`Endpoint: https://router.huggingface.co/v1/responses`);

  try {
    await createBasicAgent(
      model,
      "Explain the difference between TCP and UDP in simple terms."
    );
  } catch (error) {
    console.error("Error:", error);
    process.exit(1);
  }
}

main();

File: examples/python/basic_agent.py
"""
Basic Agent Example - Open Responses API

Demonstrates the simplest form of an agent using Open Responses
via the HuggingFace Inference Providers router.

Usage:
    pip install openai
    export HF_TOKEN=your-token
    python basic_agent.py
"""

import os
from openai import OpenAI


# Configure client with HuggingFace router endpoint
client = OpenAI(
    base_url="https://router.huggingface.co/v1",
    api_key=os.environ.get("HF_TOKEN"),
)


def create_basic_agent(model: str, input_text: str, instructions: str | None = None):
    """
    Create a basic agent request to Open Responses API.

    Args:
        model: Model identifier with provider suffix (e.g., "model:groq")
        input_text: The user's request
        instructions: Optional system prompt

    Returns:
        Response with all output items
    """
    response = client.responses.create(
        model=model,
        instructions=instructions or "You are a helpful assistant.",
        input=input_text,
    )

    return response


def display_response(response) -> None:
    """Display the response in a readable format."""
    print(f"\n{'='*60}")
    print(f"Response ID: {response.id}")
    print(f"Model: {response.model}")
    print(f"Tokens: {response.usage.input_tokens if response.usage else 0} in / {response.usage.output_tokens if response.usage else 0} out")
    print(f"{'='*60}\n")

    # Use the convenience helper for simple text output
    print("--- Output Text (convenience helper) ---")
    print(response.output_text)

    # Or iterate through all output items for more detail
    print("\n--- All Output Items ---")
    for item in response.output:
        match item.type:
            case "reasoning":
                # Open weight models provide raw content
                # Proprietary models may provide summary or encrypted_content
                text = getattr(item, "content", None) or getattr(item, "summary", None) or "[encrypted]"
                print(f"[REASONING] {text}")

            case "message":
                print(f"[MESSAGE] {getattr(item, 'content', '')}")

            case _:
                print(f"[{item.type.upper()}] {item}")


def main() -> None:
    """Main execution."""
    # Model with provider suffix - using Groq for fast inference
    model = os.environ.get("MODEL", "moonshotai/Kimi-K2-Instruct-0905:groq")
    hf_token = os.environ.get("HF_TOKEN")

    if not hf_token:
        print("Error: HF_TOKEN environment variable required")
        exit(1)

    print(f"Using model: {model}")
    print(f"Endpoint: https://router.huggingface.co/v1/responses")

    try:
        result = create_basic_agent(
            model=model,
            input_text="Explain the difference between TCP and UDP in simple terms.",
        )

        display_response(result)

    except Exception as e:
        print(f"Error: {e}")
        exit(1)


if __name__ == "__main__":
    main()
